nodestream/interpreting/interpretations/source_node_interpretation.py

```diff
@@ -108,27 +104,18 @@ def interpret(self, context: ProviderContext):
         source.properties.apply_providers(context, self.properties, self.norm_args)
         source.additional_types = self.additional_types
 
-    def gather_used_indexes(self) -> Iterable[Union[KeyIndex, FieldIndex]]:
-        # NOTE: We cannot generate indexes when we do not know the type until runtime.
-        if not self.node_type.is_static:
-            return
-
-        node_type = self.node_type.value
-        yield KeyIndex(node_type, frozenset(self.key.keys()))
-        yield FieldIndex.for_ttl_timestamp(node_type)
-        for field in self.additional_indexes:
-            yield FieldIndex(node_type, field, object_type=GraphObjectType.NODE)
+    def expand_source_node_schema(self, source_node_schema: GraphObjectSchema):
+        source_node_schema.add_keys(self.key)
+        source_node_schema.add_properties(self.properties)
+        source_node_schema.add_indexes(self.additional_indexes)
```

Comment:
This does not work when the property is not defined yet. So an add index needs to make sure the property exists.